src/iuse_actor.cpp

```diff
@@ -859,6 +859,45 @@ std::unique_ptr<iuse_actor> ups_based_armor_actor::clone() const
     return std::make_unique<ups_based_armor_actor>( *this );
 }
 
+std::unique_ptr<iuse_actor> place_npc_iuse::clone() const
+{
+    return std::make_unique<place_npc_iuse>( *this );
+}
+
+void place_npc_iuse::load( const JsonObject &obj )
+{
+    npc_class_id = string_id<npc_template>( obj.get_string( "npc_class_id" ) );
+    obj.read( "summon_msg", summon_msg );
+    obj.read( "moves", moves );
+    obj.read( "place_randomly", place_randomly );
+}
+
+int place_npc_iuse::use( player &p, item &, bool, const tripoint & ) const
+{
+    cata::optional<tripoint> target_pos;
+    if( place_randomly ) {
+        const tripoint_range target_range = points_in_radius( p.pos(), 1 );
+        target_pos = random_point( target_range, []( const tripoint & ) {
+            return true;
+        } );
+    } else {
+        const std::string query = _( "Place npc where?" );
+        target_pos = choose_adjacent( query );
```

Comment:
```suggestion
        target_pos = choose_adjacent( _( "Place npc where?" ) );
```
No need for a separate variable.